### DISCUSSION TITLE
Diagnosis Collision

### DISCUSSION THREAD

**Discussion by sampson-young** [HALRobotics/Beta#103] [Q&A]
Hi HAL,

Is there any simple example file or tutorial about the component "Diagnosis Collision"? 
I want to check the collision when the robots in operation, but don't know how this "Diagnosis Collision" works, and where should I connect the "Diagnosis Collision" output to?
For example, when moving the small brick, check the small brick and big box collision. 
![image](https://user-images.githubusercontent.com/37079727/120074411-acb26f80-c094-11eb-93d7-bcef1f677888.png)

Best wishes,
Yang

**Comment by tristangobin** [comment]
Hi Sampson-young
Thanks you to use our software.

Please find attached the demo file [CollisionCheck-Demo.zip](https://github.com/HALRobotics/Beta/files/6589980/CollisionCheck-Demo.zip) and some images to exemplifie.

![image](https://user-images.githubusercontent.com/8418539/120609664-b4866100-c452-11eb-9adf-1c76adff818c.png)

![image](https://user-images.githubusercontent.com/8418539/120609743-ca942180-c452-11eb-881b-bbdc95db1e9d.png)

![image](https://user-images.githubusercontent.com/8418539/120609899-f31c1b80-c452-11eb-974e-c75be0920dbf.png)

Best,
